include/rdma/fi_domain.h

```diff
@@ -326,6 +328,13 @@ fi_domain(struct fid_fabric *fabric, struct fi_info *info,
 	return fabric->ops->domain(fabric, info, domain, context);
 }
 
+static inline int
+fi_domain_ex(struct fid_fabric *fabric, struct fi_info *info,
+	     struct fid_domain **domain, uint64_t flags, void *context)
+{
+	return fabric->ops->domain_ex(fabric, info, domain, flags, context);
+}
```

Comment:
I'm leaning more toward calling the new functions '2' vs 'ex'.  So, fi_domain2(), which is more forward compatible if we need yet another version.

For the static inline, we need to use FI_CHECK_OP() macro to handle DL providers and providers that are not updated to use the new APIs.  That allows us to drop changes to most providers.